packages/dx-scheduler-core/src/plugins/scheduler-core/computeds.ts

```diff
@@ -17,3 +18,22 @@ export const appointments: PureComputed<
   },
   dataItem: appointment,
 }));
+
+export const dateTimeFormatComputed: any = (locale: string | string[]) => {
+  const cache = new Map();
+
+  const formatter = (nextDate: Date | string | number, nextOptions: Intl.DateTimeFormatOptions) => {
+    if (nextDate === undefined) return;
+    const date = (typeof nextDate) === 'object' ? nextDate : new Date(nextDate);
+    const key = JSON.stringify(nextOptions);
+
+    if (cache.has(key)) {
+      return cache.get(key).format(date);
+    }
+
+    const formatInstance = dateTimeFormatInstance(locale, nextOptions);
+    cache.set(key, formatInstance);
+    return formatInstance.format(date);
```

Comment:
Cache management and formatting is mixed.

```js
if (!cache.has(key)) {
  const value = ...
  cache.set(key, value);
}

return cache.get(key).format(date);
```

Reply:
But in this case, we excess call `get` method if `!cache.has(key) === true`

Reply:
`get` call is much cheaper than instantiating a dateTime formatter each time - that's the purpose of caching :)

Reply:
There is always one `get` call - one `has`, one or zero `set`, one `get`.

Reply:
In my case, I call only `one has + one set` or `one has + one get`

Reply:
> instantiating a dateTime formatter each time

Turned out it wasn't. My mistake, I haven't seen a whole function code

Reply:
`get` call is avoided in the case when cache missed - its probability is much less than that when cache is not missed (and `get` is not avoided).
There is not much point in avoiding `get` in a rare case only.

Anyway, caching and formatting still can be splitted.

```js
let formatInstance = cache.has(key) ? cache.get(key) : null;
if (!formatInstance) {
  formatInstance = createIt();
  cache.set(key, formatInstance);
}

return formatInstance.format(date);
```

Further, `has` can be removed because `formatInstance` cannot be falsy.

```js
let formatInstance = cache.get(key);
if (!formatInstance) { ... }
```

Reply:
Done